src/render/PointArray.cpp

```diff
@@ -386,14 +386,83 @@ bool PointArray::loadFile(QString fileName, size_t maxPointCount)
     {
         g_logger.debug("Reordering field %d: %s", i, m_fields[i]);
         reorder(m_fields[i], inds.get(), m_npoints);
-        emit loadProgress(int(100*(i+1)/m_fields.size()));
+        emit loadProgress(int(100*(i+1)/(m_fields.size()+1))); // denominator +1 for permutation reorder below
     }
     m_P = (V3f*)m_fields[m_positionFieldIdx].as<float>();
+    // The index we want to store is the reverse permutation of the index above
+    m_inds = std::unique_ptr<size_t[]>(new size_t[m_npoints]);
```

Comment:
Should we make this `uint32_t`, given that you're assuming that elsewhere anyway?